### PR TITLE
Dashboard hotkeys

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes
- Continues #3608
- This PR introduces hotkey navigation for the dashboards.
- Introduces `HotkeyButton` which extends Ant's button to display a hotkey indicator as outlined on Figma.
    
- Introduces `KeyboardHotkeys` component to easily bind hotkeys in a view to specific actions.
- Fully refactors the dashboards multiple modes (`isOnEditMode`, `isOnPresentationMode`, ....) into a single one (`dashboardMode`). This not only greatly simplifies the logic and code readability but enforces that a user can only be in one at a time. Previously you could be for instance in full screen and edit mode.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

File: frontend/src/lib/components/KeyboardHotkeys.tsx
Comment (paolodamico) [inline frontend/src/lib/components/KeyboardHotkeys.tsx:11]: @mariusandra would love your input here. As you can see the keys are being repeated manually but are already registered as a type `Keys` on `~/types.ts`, is there a way to use that type here without having to repeat everything?

**Comment by paolodamico** [comment]
Please don't merge this after reviewing. Need to merge the parent PR first and rebase.

File: frontend/src/lib/components/KeyboardHotkeys.tsx
Comment (Twixes) [reply frontend/src/lib/components/KeyboardHotkeys.tsx]: Indeed there is :) Take a look now

File: frontend/src/lib/components/KeyboardHotkeys.tsx
Comment (Twixes) [inline frontend/src/lib/components/KeyboardHotkeys.tsx:11]: Since this doesn't actually return any JSX element, should be a React hook instead

**Review by Twixes — CHANGES_REQUESTED** [review]
Only took a very cursory glance: this shadow looks much out of place IMO

shadow

File: frontend/src/lib/components/KeyboardHotkeys.tsx
Comment (paolodamico) [reply frontend/src/lib/components/KeyboardHotkeys.tsx:11]: Right! Will make that change

File: frontend/src/lib/components/KeyboardHotkeys.tsx
Comment (paolodamico) [reply frontend/src/lib/components/KeyboardHotkeys.tsx]: awesome! thanks Michael! so simple in hindsight 🙈

File: frontend/src/lib/components/KeyboardHotkeys.tsx
Comment (paolodamico) [reply frontend/src/lib/components/KeyboardHotkeys.tsx]: As discussed on Zoom, reverting back to functional component instead of hook as it was causing some obscure React error.

**Comment by paolodamico** [comment]
Ok rebased and adjusted, ready for one final look @Twixes

**Review by Twixes — APPROVED** [review]

**Comment by paolodamico** [comment]
Thanks for the review and the updates to the PR @Twixes! Will give it one quick look and merge.

**Pull request merged**
